tsqa/test_cases.py

```diff
@@ -46,7 +46,7 @@ class EnvironmentCase(unittest.TestCase):
     def run(self, result=None):
         unittest.TestCase.run(self, result)
         # we want to keep track of failures at a class level-- not instance level
-        self.__class__.__successful &= result.result.wasSuccessful()
+        self.__class__.__successful &= result.wasSuccessful()
```

Comment:
Are you seeing exceptions because of this? result.result is actually correct (at least in my testing).